packages/dds/matrix/src/test/matrix.undo.spec.ts

```diff
@@ -574,29 +574,41 @@ describe("Matrix", () => {
 					[2, 3],
 				]);
 
+				// now undo both inserts
 				undo2.undoOperation();
 				await expect([[0, 1]]);
 
 				undo1.undoOperation();
 				await expect([]);
 
+				// redo re-inserts in opposite order to their original
+				// client 2, then client 1
 				undo2.redoOperation();
 				await expect([[2, 3]]);
 
 				undo1.redoOperation();
 				await expect([
-					[0, 1],
 					[2, 3],
+					[0, 1],
 				]);
 
+				// now undo both inserts again
 				undo1.undoOperation();
 				await expect([[2, 3]]);
 
-				undo1.undoOperation();
-				await expect([[]]);
+				undo2.undoOperation();
+				await expect([]);
 
+				// redo again in the opposite order to switch row order
+				// client 1, then client 2
 				undo1.redoOperation();
-				await expect([[2, 3]]);
+				await expect([[0, 1]]);
+
+				undo2.redoOperation();
+				await expect([
+					[0, 1],
```

Comment:
ending order of row is reserved from above since the local undo redos happen in the opposite order.